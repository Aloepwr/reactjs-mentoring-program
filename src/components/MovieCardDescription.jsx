import React from 'react';
import { withRouter } from 'react-router';

import "../stylesheets/MovieCardDescription.less";


const MovieCardDescription = withRouter( (props) => {
  const { poster, title, vote_average, release_date, runtime, overview, budget, seasons, last_air_date } = props.currentMovie;
  
  const rating = <p>{ vote_average }</p>;

  return (
    <div className="movie-description">
      <div className="movie-description__head">
        <span className="span-title">netflixroulette</span>
        <button className="btn__back" onClick={ () => props.history.goBack() }>Search</button>
      </div>
      
      <div className="movie-description__layout">
        <div className="movie-description__image">
          { poster ? <img src={`https://image.tmdb.org/t/p/w185_and_h278_bestv2${poster}`} /> : null }
        </div>
        <div className="movie-description__details">
          <div className="movie-description__details--title">
            <h2>{ title }</h2>
            { vote_average ? rating : null }
          </div>
          <div className="movie-description__info">
            { release_date ? <p>{release_date}</p> : null }
            { runtime ? <p>{runtime} min</p> : null }
          </div>
          <p className="movie-description__summary">{overview}</p>
          <div className="movie-description__bottom-info">
            { budget ? <p>Budget is {budget}$</p> : null }
            { seasons ? <p>Seasons: {seasons}</p> : null }
            { last_air_date ? <p>Last air date: {last_air_date}</p> : null }
          </div>
        </div>
      </div>
    </div>
  );
});

export default MovieCardDescription;
