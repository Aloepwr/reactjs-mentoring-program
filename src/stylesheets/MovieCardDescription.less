//
// movie-description
//

.movie-description {
  width: 980px;
  padding: 25px 40px 40px;
  margin-left: auto;
  margin-right: auto;
  margin-bottom: 80px;
  border-radius: 5px;
  
  background-color: rgba(0, 0, 0, 0.85);

  font-family: Arial;
  color: #DBD3D6;

  p {
    padding: 10px 0;
  }
}

.movie-description__head {
  display: flex;
  justify-content: space-between;
  align-items: center;

  margin-bottom: 35px;
  padding: 0 30px;

  .btn__back {
    width: 100px;
    height: 30px;

    border: 0;
    border-radius: 2px;
    background-color: white;
    text-transform: uppercase;
    font-size: 0.9em;
    font-weight: bold;
    letter-spacing: -0.2px;
    color: red;

    cursor: pointer;
  }
}

.movie-description__layout {
  display: flex;
  justify-content: space-between;
}

.movie-description__image {
  width: 37%;
}

.movie-description__details {
  width: 63%;
  padding: 5px;
}

.movie-description__details--title {
  display: flex;
  align-items: center;

  h2 {
    color: #D6064F;
    font-size: 2.75em;
  }

  p {
    margin-left: 20px;
    padding: 16px 13px;

    border: 1px solid green;
    border-radius: 50%;
    
    font-size: 16px;
    color: green;
  }
}

.movie-description__info {
  display: flex;

  margin-bottom: 20px;

  font-size: 1.1em;
  font-weight: bold;
  color: #fff;

  p {
    margin-right: 25px;

    &:last-child {
      margin-right: 0;
    }
  }
}

.movie-description__summary {
  margin-bottom: 20px;
  width: 90%;

  font-size: 1.15em;
  line-height: 1.15em;
}

.movie-description__bottom-info {
  width: 90%;
  
  font-size: 0.85em;
  line-height: 1.2em;
}