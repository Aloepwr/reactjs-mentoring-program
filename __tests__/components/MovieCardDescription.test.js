import React from 'react';
import renderer from "react-test-renderer";
import { MemoryRouter } from 'react-router-dom';
import MovieCardDescription from "../../src/components/MovieCardDescription";

const movie = {
  title: "test title",
  release_date: "1906-08-24",
  poster: "/wajNlrVDkryVGTzDz9vZE6XKZnC.jpg",
  id: 111,
  type: "movie",
  vote_average: 4.1,
  runtime: 150,
  overview: "Some overview",
  budget: 10000,
  seasons: 1,
  last_air_date: "2074-11-27"
};

describe("Movie card description component", () => {
  test("renders valid layout when it receives all props", () => {
    const tree = renderer.create(
      <MemoryRouter>
        <MovieCardDescription currentMovie={ movie }/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  const movieWithoutSomeProps = {
    title: "test title",
    poster: "/wajNlrVDkryVGTzDz9vZE6XKZnC.jpg",
    id: 111,
    type: "movie",
    overview: "Some overview"
  };

  test("renders layout without vote_average, release_date, runtime, budget, seasons, last_air_date ", () => {
    const tree = renderer.create(
      <MemoryRouter>
        <MovieCardDescription currentMovie={ movieWithoutSomeProps }/>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});