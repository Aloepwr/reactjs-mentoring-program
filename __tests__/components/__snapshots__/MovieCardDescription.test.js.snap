// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie card description component renders layout without vote_average, release_date, runtime, budget, seasons, last_air_date  1`] = `
<div
  className="movie-description"
>
  <div
    className="movie-description__head"
  >
    <span
      className="span-title"
    >
      netflixroulette
    </span>
    <a
      className="btn__back"
      href="/search"
      onClick={[Function]}
    >
      Search
    </a>
  </div>
  <div
    className="movie-description__layout"
  >
    <div
      className="movie-description__image"
    >
      <img
        src="https://image.tmdb.org/t/p/w185_and_h278_bestv2/wajNlrVDkryVGTzDz9vZE6XKZnC.jpg"
      />
    </div>
    <div
      className="movie-description__details"
    >
      <div
        className="movie-description__details--title"
      >
        <h2>
          test title
        </h2>
      </div>
      <div
        className="movie-description__info"
      />
      <p
        className="movie-description__summary"
      >
        Some overview
      </p>
      <div
        className="movie-description__bottom-info"
      />
    </div>
  </div>
</div>
`;

exports[`Movie card description component renders valid layout when it receives all props 1`] = `
<div
  className="movie-description"
>
  <div
    className="movie-description__head"
  >
    <span
      className="span-title"
    >
      netflixroulette
    </span>
    <a
      className="btn__back"
      href="/search"
      onClick={[Function]}
    >
      Search
    </a>
  </div>
  <div
    className="movie-description__layout"
  >
    <div
      className="movie-description__image"
    >
      <img
        src="https://image.tmdb.org/t/p/w185_and_h278_bestv2/wajNlrVDkryVGTzDz9vZE6XKZnC.jpg"
      />
    </div>
    <div
      className="movie-description__details"
    >
      <div
        className="movie-description__details--title"
      >
        <h2>
          test title
        </h2>
        <p>
          4.1
        </p>
      </div>
      <div
        className="movie-description__info"
      >
        <p>
          1906-08-24
        </p>
        <p>
          150
           min
        </p>
      </div>
      <p
        className="movie-description__summary"
      >
        Some overview
      </p>
      <div
        className="movie-description__bottom-info"
      >
        <p>
          Budget is 
          10000
          $
        </p>
        <p>
          Seasons: 
          1
        </p>
        <p>
          Last air date: 
          2074-11-27
        </p>
      </div>
    </div>
  </div>
</div>
`;
